client/my-sites/themes/validate-filters.js

```diff
@@ -30,5 +31,21 @@ module.exports = function validateFilter( context, next ) {
 	}
 
 	next();
-};
+}
 
+export function validateVertical( context, next ) {
+	const { vertical } = context.params;
+
+	if ( ! vertical ) {
+		return next();
+	}
+
+	if ( ! includes( getSubjects(), vertical ) ) {
+		if ( context.isServerSide ) {
+			return context.res.redirect( '/themes' );
+		}
+		return page.redirect( '/themes' );
```

Comment:
Ideally, yeah. Unless we need to fall through this route def and have another one catch it in case we need it for the v4 redirects (i.e. we need to check the routes listed in https://github.com/Automattic/wp-calypso/pull/12768).